tinygrad/runtime/ops_disk.py

```diff
@@ -1,26 +1,15 @@
-import os, mmap, _posixshmem, io, functools
-from typing import Dict, List, Any
-from tinygrad.dtype import DType, dtypes
-from tinygrad.helpers import prod, OSX
-from tinygrad.device import Compiled, Allocator, JITRunner, Buffer
-from tinygrad.ops import UnaryOps, LazyOp, BufferOps
-from tinygrad.shape.view import strides_for_shape
-
-class UnderlyingDiskBuffer:
-  def __init__(self, fd, mem): self.fd, self.mem = fd, mem
-  def __del__(self):
-    if self.fd is not None: os.close(self.fd)
+import os, mmap, _posixshmem, io
+from tinygrad.helpers import OSX
+from tinygrad.device import Compiled, Allocator
 
 class DiskBuffer:
-  def __init__(self, ud:UnderlyingDiskBuffer, size:int, dtype:DType=dtypes.uint8, offset=0):
-    self.ud, self.size, self.dtype, self.offset = ud, size, dtype, offset
-  def __repr__(self): return f"<DiskBuffer size={self.size} dtype={self.dtype} offset={self.offset}>"
-  def _buf(self) -> memoryview: return memoryview(self.ud.mem)[self.offset:self.offset+self.size*self.dtype.itemsize]
+  def __init__(self, fd, mem, offset, size): self.fd, self.mem, self.offset, self.size = fd, mem, offset, size
+  def _buf(self) -> memoryview: return memoryview(self.mem)[self.offset:self.offset+self.size]
 
 MAP_LOCKED, MAP_POPULATE = 0 if OSX else 0x2000, getattr(mmap, "MAP_POPULATE", 0 if OSX else 0x008000)
 class DiskAllocator(Allocator):
   def __init__(self, device:str): self.device = device
-  def _alloc(self, size:int, options):
+  def _alloc(self, size:int, options) -> DiskBuffer:
```

Comment:
You have to be careful that this isn't being called multiple times for the same file, you don't want it reopening the file a ton of times.